utils/dwarf.c

```diff
@@ -1016,6 +997,69 @@ static int get_dwarfspecs_cb(Dwarf_Die *die, void *data)
 	return DWARF_CB_OK;
 }
 
+struct cu {
+	char* name;
+	int n;
+};
+
+int map_put(struct cu *cands, int n_cands, char *name, int n) {
+	struct cu *c = &cands[n_cands];
+	c->name = xstrdup(name);
+	c->n = 1;
+	return 0;
+}
+
+struct cu* map_get(struct cu *cands, int n_cands, char *name) {
+	int i = 0;
+	for (i = 0; i < n_cands; i++) {
+		if (!strcmp(cands[i].name, name))
+			return &cands[i];
+	}
+	return NULL;
+}
+
+static int func_cmp(const void *a, const void *b)
+{
+	const char *fa = a;
+	const char *fb = b;
+
+	return strcmp(fa, fb);
+}
+
+char* get_base_dir(char* cus[], int n_cu) {
+	struct cu *cands = (struct cands*)malloc(sizeof(struct cu) * n_cu);
+	struct cu *cand;
+	int n_cands = 0;
+
+	qsort(cus, n_cu, sizeof(char*), func_cmp);
+
+	for (int i = 0; i < n_cu; i++) {
+		char *cu_prefix = xstrdup(cus[i]);
+		cand = NULL;
+		while (cu_prefix && *cu_prefix) {
+			struct cu *cand = map_get(cands, n_cands, cus[i]);
+			if (cand) {
+				cand->n++;
+				break;
```

Comment:
I'm afraid that it depends on the timing when a shorter entry appears.  So I think it's better collect it with full name, and then calculate/aggregate counts at last.